src/librustc_unicode/char.rs

```diff
@@ -448,21 +703,36 @@ impl char {
         }
     }
 
-    /// Converts a character to its lowercase equivalent.
+    /// Returns an iterator that yields the lowercase equivalent of a `char`.
     ///
-    /// This performs complex unconditional mappings with no tailoring.
-    /// See `to_uppercase()` for references and more information.
+    /// If no conversion is possible then an iterator with just the input character is returned.
     ///
-    /// # Return value
+    /// This performs complex unconditional mappings with no tailoring: it maps
+    /// one Unicode character to its lowercase equivalent according to the
+    /// [Unicode database] and the additional complex mappings
+    /// [`SpecialCasing.txt`]. Conditional mappings (based on context or
+    /// language) are not considered here.
     ///
-    /// Returns an iterator which yields the characters corresponding to the
-    /// lowercase equivalent of the character. If no conversion is possible then
-    /// an iterator with just the input character is returned.
+    /// For a full reference, see [here][reference].
+    ///
+    /// [Unicode database]: ftp://ftp.unicode.org/Public/UNIDATA/UnicodeData.txt
+    ///
+    /// [`SpecialCasing.txt`]: ftp://ftp.unicode.org/Public/UNIDATA/SpecialCasing.txt
+    ///
+    /// [reference]: http://www.unicode.org/versions/Unicode7.0.0/ch03.pdf#G33992
     ///
     /// # Examples
     ///
+    /// Basic usage:
+    ///
     /// ```
-    /// assert_eq!(Some('c'), 'C'.to_lowercase().next());
+    /// let c = 'c';
+    ///
+    /// assert_eq!(c.to_uppercase().next(), Some('C'));
+    ///
+    /// // 日本語 does not have case
```

Comment:
One difficulty is that 漢字 (the name of the writing system / script) is romanized differently depending on which language’s pronunciation you use. So maybe like this, to avoid picking sides?

```rust
/// // 漢字 (Kanji / Hanji) does not have case
```